test/hotspot/jtreg/runtime/locking/TestRecursiveMonitorChurn.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import jdk.test.lib.process.OutputAnalyzer;
+import jdk.test.lib.process.ProcessTools;
+
+import java.io.IOException;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/*
+ * @test
+ * @summary Tests that recursive locking doesn't cause excessive native memory usage
+ * @library /test/lib
+ * @run main TestRecursiveMonitorChurn
```

Comment:
Better to use 
@run driver TestRecursiveMonitorChurn
for tests which fork VM.

Reply:
Ok, did that. Can you explain to me what difference does it make?

Reply:
The 'driver' code is executed without any additional VM options. So it just runs faster then main.
See doc: https://openjdk.org/jtreg/tag-spec.html

`
driver[/fail][/timeout=<seconds>] <class> <arg>*
Invoke the main method of the specified class, passing any arguments after the class name. Although superficially similar to @run main, this is for use when it is desirable to perform additional setup or configuration before running the class containing the actual test code, possibly in a child VM.

The named <class> will be compiled on demand, just as though an "@run build <class>" action had been inserted before this action. If this action requires classes other than <class> to be up to date, insert an appropriate build action before this action.
`